### PR TITLE
Analytics: skip requests from bots on 404s

### PR DESCRIPTION
**Pull request by stsewd**
We need to create a transform rule on cloudflare with this dynamic header on the request. I have already created that rule for a test domain.

Closes https://github.com/readthedocs/readthedocs.org/issues/9116

### CODE REVIEW

File: readthedocs/proxito/views/serve.py [hunk 1]
```diff
@@ -406,6 +406,18 @@ def _register_broken_link(self, project, version, path, full_path):
             if not project.has_feature(Feature.RECORD_404_PAGE_VIEWS):
                 return
 
+            # This header is set from Cloudflare,
+            # it goes from 0 to 100, 0 being low risk,
+            # and values above 10 are bots/spammers.
```
Comment (ericholscher) [inline readthedocs/proxito/views/serve.py:411]: Great comment. 👍

File: readthedocs/proxito/views/serve.py [hunk 2]
```diff
@@ -406,6 +406,18 @@ def _register_broken_link(self, project, version, path, full_path):
...
+            # https://developers.cloudflare.com/ruleset-engine/rules-language/fields/#dynamic-fields.
+            threat_score = int(self.request.headers.get('X-Cloudflare-Threat-Score', 0))
+            if threat_score > 10:
+                log.info(
```
Comment (ericholscher) [inline readthedocs/proxito/views/serve.py:415]: I imagine we'll need to drop this to a `log.debug` at some point, but this is fine for now.

**Review by ericholscher — APPROVED** [review]
This looks 💯. We should enable this for our domains and see what we get.

File: readthedocs/proxito/views/serve.py [hunk 3]
```diff
@@ -406,6 +406,18 @@ def _register_broken_link(self, project, version, path, full_path):
             if not project.has_feature(Feature.RECORD_404_PAGE_VIEWS):
                 return
 
+            # This header is set from Cloudflare,
+            # it goes from 0 to 100, 0 being low risk,
+            # and values above 10 are bots/spammers.
+            # https://developers.cloudflare.com/ruleset-engine/rules-language/fields/#dynamic-fields.
+            threat_score = int(self.request.headers.get("X-Cloudflare-Threat-Score", 0))
```
Comment (humitos) [inline readthedocs/proxito/views/serve.py:413]: We may need to handle the exception here in case the string is not able to be converted to int.

Comment (humitos) [reply readthedocs/proxito/views/serve.py:413]: Cool! 👍🏼

**Pull request merged**